estuary/ui
# Expose `onIncompatibleSchemaChange`  properties in materialization workflows

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/1375

## Changes

### 1375

The following features are included in this PR:

* Enable the specification-level `onIncompatibleSchemaChange` property to be configured in materialization workflows. An autocomplete field located in the top-level _Backfill_ section accomplishes this.

* Create a hook which updates the specification-level `onIncompatibleSchemaChange` property.

* Extend `BindingState` with an `onIncompatibleSchemaChange` property that captures the value of the specification-level `onIncompatibleSchemaChange` property.

* Create a utility function that updates or removes the specification-level `onIncompatibleSchemaChange` property.

* Enable the binding-level `onIncompatibleSchemaChange` property to be configured in materialization workflows. An autocomplete field located in the binding-level _Backfill_ section accomplishes this.

* Rename the hook which updates the binding-level `onIncompatibleSchemaChange` property.

* Extend `BindingState` with a `ResourceConfig['meta']['onIncompatibleSchemaChange']` property that captures the value of the binding-level `onIncompatibleSchemaChange` property.

## Tests

### Manually tested

Approaches to testing are as follows:

* Validate that the incompatible schema change section does not appear in capture workflows.

* Validate that the field initialized properly when editing a materialization that has the specification-level, `onIncompatibleSchemaChange` property set to a supported value.

* Validate that the drafted specification is updated properly when the field is set to a supported value and cleared.

### Automated tests

N/A

#### Playwright tests ran locally


## Screenshots

This field is positioned between the top-level _Backfill_ section and the binding selector.

**Spec-level, `onIncompatibleSchemaChange` field | Empty**

<img width="413" alt="pr_screenshot-1372-incompatible_schema-spec_level-empty" src="https://github.com/user-attachments/assets/b9183a14-ca24-4d02-9d9c-29869f36d73c">

<br />
<br />

**Spec-level, `onIncompatibleSchemaChange` field | Menu options**

<img width="253" alt="pr_screenshot-1372-incompatible_schema-spec_level-menu_options" src="https://github.com/user-attachments/assets/d51a32b8-7971-495a-8836-0896fe9f9863">

<br />
<br />

**Spec-level, `onIncompatibleSchemaChange` field | Option selected**

<img width="416" alt="pr_screenshot-1372-incompatible_schema-spec_level-option_selected" src="https://github.com/user-attachments/assets/a4899d44-f07f-44cf-bf04-70501b5c6da1">

<br />
<br />

**Spec-level, `onIncompatibleSchemaChange` field | Unsupported value detected | String**

<img width="326" alt="pr_screenshot-1372-incompatible_schema-spec_level-error-invalid_existing_value" src="https://github.com/user-attachments/assets/3f1bce5b-2779-4edc-9b2c-1a35098c6e26">

<br />
<br />

**Spec-level, `onIncompatibleSchemaChange` field | Unsupported value detected | Other**

<img width="353" alt="pr_screenshot-1372-incompatible_schema-spec_level-error-unsupported_value-object" src="https://github.com/user-attachments/assets/14378e6b-1b96-4e76-a313-0cd61546b075" />
